src/dbnode/ts/types.go

```diff
@@ -94,8 +94,9 @@ type Series struct {
 
 // A Datapoint is a single data value reported at a given time.
 type Datapoint struct {
-	Timestamp time.Time
-	Value     float64
+	Timestamp      time.Time
+	TimestampNanos int64
```

Comment:
Use `xtime.UnixNanos`?

Reply:
PS do we need both timestamp and the nanos?  Can we just have `xtime.UnixNanos` here?  Or is there too many places calling `.Timestamp` that would need updating?

Reply:
Yeah we thought that since there are so many references to `.Timestamp`, it'd be worth scoping this change to just the relevant areas we are optimizing on the read-side. Open to changing our plan on this though